Document v3 custom partitions (#5309)
influxdata/docs-v2: content/influxdb/cloud-dedicated/admin/custom-partitions/partition-templates.md

---
title: Partition templates
list_title: Use partition templates
description: >
  Learn how to define custom partitioning strategies using partition templates.
  Data can be partitioned by tag and time.
menu:
  influxdb_cloud_dedicated:

>     parent: Manage data partitioning

Reviewer: ```suggestion
    name: Partition templates
    parent: Manage data partitioning
```

weight: 202
---

> Use partition templates to define the patterned used to generate partition keys.

Reviewer: ```suggestion
Use partition templates to define the patterns used to generate partition keys.
```

A partition key uniquely identifies a partition and is used to name the partition

> Parquet file in [Object storage](/influxdb/cloud-dedicated/reference/internals/storage-engine/#object-storage).

Reviewer: ```suggestion
Parquet file in the [Object store](/influxdb/cloud-dedicated/reference/internals/storage-engine/#object-store).
```

A partition template consists of 1-8 _template parts_---dimensions to partition data by.
There are two types of parts:

- **tag**: [InfluxDB tag](/influxdb/cloud-dedicated/reference/glossary/#tag) to
  partition by.

>   _A partition template can include up to 7 tag parts._

Reviewer: ```suggestion
  _A partition template can include up to seven tag parts._
```

- **time**: A Rust strftime date and time string that specifies the time interval
  to partition data by. The smallest unit of time included in the time part
  template is the interval used to partition data.
  _A partition template includes only 1 time part._

<!-- TOC -->
- [Restrictions](#restrictions)
  - [Template part size limit](#template-part-size-limit)
  - [Reserved keywords](#reserved-keywords)
  - [Reserved Characters](#reserved-characters)
- [Tag part templates](#tag-part-templates)
- [Time part templates](#time-part-templates)
  - [Date specifiers](#date-specifiers)
  - [Time specifiers](#time-specifiers)
  - [Time zone specifiers](#time-zone-specifiers)
  - [Date and time specifiers](#date-and-time-specifiers)
  - [Special specifiers](#special-specifiers)
<!-- /TOC -->

## Restrictions

### Template part size limit

Each template part is limited to 200 bytes in length.
Anything longer will be truncated at 200 bytes and appended with `#`.

### Reserved keywords

The following reserved keywords cannot be used in partition templates:

- `time`

### Reserved Characters

If used in template parts, non-ASCII characters and the following reserved

> characters must be percent encoded:

Reviewer: ```suggestion
characters must be [percent encoded](https://developer.mozilla.org/en-US/docs/Glossary/Percent-encoding):
```

|  `%#z`   | `+09`                              | *Parsing only:* Same as `%z` but allows minutes to be missing or present.                                          |

### Date and time specifiers

| Variable | Example                            | Description                                                            |
| :------: | :--------------------------------- | :--------------------------------------------------------------------- |
|   `%c`   | `Sun Jul  8 00:34:60 2001`         | Locale's date and time (e.g., Thu Mar  3 23:05:25 2005).               |
|   `%+`   | `2001-07-08T00:34:60.026490+09:30` | ISO 8601 / RFC 3339 date & time format. [^5]                           |
|   `%s`   | `994518299`                        | UNIX timestamp, the number of seconds since 1970-01-01 00:00 UTC. [^6] |

### Special specifiers

| Variable | Example | Description             |
| :------: | :------ | :---------------------- |
|   `%t`   |         | Literal tab (`\t`).     |
|   `%n`   |         | Literal newline (`\n`). |
|   `%%`   |         | Literal percent sign.   |

It is possible to override the default padding behavior of numeric specifiers `%?`.

> This is not allowed for other specifiers and will result in the `BAD_FORMAT` error.

Reviewer: ```suggestion
This is not allowed for other specifiers and results in the `BAD_FORMAT` error.
```
